apps/src/templates/courseOverview/ResourcesEditor.js

```diff
@@ -57,11 +78,14 @@ export default class ResourcesEditor extends Component {
   }
 
   render() {
-    const { resources } = this.state;
+    const { resources, errorString } = this.state;
+
+    // avoid showing multiple empty reosurces
+    const lastNonEmpty = _.findLastIndex(resources, ({type, link}) => link && type);
 
     return (
       <div>
-        {resources.map((resource, index) =>
+        {resources.slice(0, lastNonEmpty + 2).map((resource, index) =>
```

Comment:
If we have 2 non-empty resources, we want to show 3 resources (so that the user can add more). So + 1 for the additional non-empty, and +1 for slice. I'll add a comment to make this clearer.